### ISSUE TITLE
Refactor optimization part (gradient descent, backprop and normalization)

### ISSUE THREAD

**Issue by louishhy** [ut-vision/30min-of-deep-learning#20]
The optimization topic contains many important hands-ons,
the current implementation does not cover all of them.

## Possible proposal
Refactor to `Optimization` since it is all related to optimization.

To comply with the 30min limit it is possible to split into multiple parts, like 05a, 05b, etc.
Can separate by: Ch. 05 regarding optimization, Ch. 06 Regarding regularization.
Parts that are covered are marked with [OK].

05 - Optimization
[OK] Auto-differentiation engine in PyTorch 
[OK] Gradient descent and PyTorch optimizer in practice
[] Momentum
[] Parameter initialization, Xavier and Kaiming initialization and corresponding activation functions
[] Learning rate scheduling
[] Normalization: Data normalization, Batch normalization and Layer normalization
[] Problems with deep networks, residual connections, and ResNet.

===

06 - Regularization
[OK] L1 / L2 regularization, dropout
